st2common/st2common/services/packs.py

```diff
@@ -215,3 +218,43 @@ def search_pack_index(
                 break
 
     return list(itertools.chain.from_iterable(matches))
+
+
+def delete_action_files_from_pack(pack_name, entry_point, metadata_file):
+    """
+    Prepares the path for entry_point file and metadata file of action and
+    deletes them from disk.
+    """
+    pack_base_path = get_pack_base_path(pack_name=pack_name)
+    action_entrypoint_file_path = os.path.join(pack_base_path, "actions", entry_point)
+    action_metadata_file_path = os.path.join(pack_base_path, metadata_file)
+
+    if os.path.exists(action_entrypoint_file_path):
+        try:
+            os.remove(action_entrypoint_file_path)
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_entrypoint_file_path,
+            )
+        except Exception as e:
+            LOG.error(
+                'Unable to delete "%s" file. Exception was "%s"',
+                action_entrypoint_file_path,
+                e,
+            )
+
+    if os.path.exists(action_metadata_file_path):
+        try:
+            os.remove(action_metadata_file_path)
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_metadata_file_path,
+            )
+        except Exception as e:
+            LOG.error(
```

Comment:
If we fail to delete the files from disk as we are catching the exceptions, does this not mean that the API is going to return success?
If we fail to delete from the database then the action appears to be to log.error, then in the API we return internal_server_error. But we can't do that if we suppress the exception here.


Would it be better to re-throw the exception so that in the api/controllers you can then return an internal server error?